ScoutSuite/providers/aws/resources/resources.py

```diff
@@ -24,29 +23,49 @@ def __init__(self, facade, scope: dict):
 
 class AWSCompositeResources(AWSResources, CompositeResources, metaclass=abc.ABCMeta):
 
-    """This class represents a collection of AWSResources. Classes extending AWSCompositeResources should \
-    define a "_children" attribute which consists of a list of tuples describing the children. The tuples \
-    are expected to respect the following format: (<child_class>, <child_name>). The child_name is used by \
-    indicates the name under which the child will be stored in the parent object.
+    """This class represents a collection of AWSResources. Classes extending AWSCompositeResources should define a
+    "_children" attribute which consists of a list of tuples describing the children. The tuples are expected to
+    respect the following format: (<child_class>, <child_name>). The child_name is used by indicates the name under
+    which the child will be stored in the parent object.
     """
-        
-    async def _fetch_children(self, parent: object, scope: dict):
-        """This method calls fetch_all on each child defined in "_children" and stores the fetched resources \
-        in the parent under the key associated with the child. It also creates a "<child_name>_count" entry \
-        for each child.
 
-        :param parent: The object in which the children should be stored
-        :param scope: The scope passed to the children constructors
+    async def _fetch_children_of_all_resources(self, resources: dict, scopes: dict):
```

Comment:
Should that be made _public_ (aka remove the `_`)? It seems like it's used outside of this class.

Reply:
Where? This method is always called from a child class of `AWSCompositeResources` (inside the `fetch_all` method), like `_fetch_children` and thus, never used outside of this class.

Reply:
Oh you're right. I didn't realize they were all child classes. All good!